jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -248,6 +269,11 @@ public int read() throws IOException
 
     @Override
     public int read(byte[] b, int off, int len) throws IOException
+    {
+        return read(ByteBuffer.wrap(b, off, len));
```

Comment:
The main flow in read(byte[] b, int off, int len)  is same as read(ByteBuffer buffer).  Wrap the array can avoid these duplicate code.